src/metabase/sync/analyze/query_results.clj

```diff
@@ -61,27 +62,26 @@
   "Return the desired storage format for the column metadata coming back from RESULTS and fingerprint the RESULTS."
   [results]
   (let [result-metadata (for [col (:cols results)]
-                          (-> col
-                              stored-column-metadata->result-column-metadata
-                              (maybe-infer-special-type col)))]
+                          ;; Rarely certain queries will return columns with no names. For example
+                          ;; `SELECT COUNT(*)` in SQL Server seems to come back with no name.
+                          ;; Since we can't use those as field literals in subsequent queries,
+                          ;; filter them out.
+                          (when-not (str/blank? (:name col))
```

Comment:
Zooming in at this point I'm not 100% on this, but it looks like there are a lot of gymnastics around `nil`s. Specially this line, line 76 and line 84. Also looks like those `nil`s that are being dodged are just removed at the end. Move this `when-not` to a `:when (not (str/blank? (:name col))` on the `for` and avoid all of the nil checks downstream?

Reply:
We can't totally skip those columns yet, because if we do the number of fingerprinters and their positions won't match the columns in the resultset (for efficiency sake `col-wise` requires a transducer for each column).